src/components/keyboard-nav.js

```diff
@@ -0,0 +1,92 @@
+import _ from 'lodash/fp'
+import { useRef } from 'react'
+import * as Utils from 'src/libs/utils'
+
+
+export const withArrowKeyNavigation = ({ onKeyDown, children }) => {
+  const count = children.length
+  const refs = _.map(useRef, _.range(0, count))
+
+  const focusOn = index => {
+    // Wrap around the ends and ensure the number is positive
+    const i = index % count + (index < 0 ? count : 0)
+    refs[i].current?.focus()
+  }
+
+  // Pass a ref into each child, using an alternative property name
+  return _.map(([i, child]) => {
```

Comment:
Then if we wrap this in a div, I think the right thing to do might be to wrap this in a Fragment, so we're not passing the result of a map directly to a dom element.